test/system/users/edit_user_test.rb

```diff
@@ -163,4 +163,17 @@ def test_profile_page_has_a_link_to_terms_and_conditions
     visit user_path(@user.uuid)
     assert(page.has_link?('Review terms and condition', href: Setting.terms_and_conditions_link))
   end
+
+  def test_user_gets_mail_when_password_changed
+    visit edit_user_path(@user.uuid)
+    fill_in('user[password]', with: 'newpassword')
+    fill_in('user[password_confirmation]', with: 'newpassword')
+    fill_in('user[current_password]', with: 'password123')
+    assert(page.has_button?('Update'))
+    click_link_or_button('Update')
+    
+    last_email = ActionMailer::Base.deliveries.last
+    assert_equal('Password Changed', last_email.subject)
+    assert_equal(['user@auction.test'], last_email.to)
```

Comment:
In general, such things are tested in mailer tests, but in this particular case I wonder if it's needed to be tested at all, since it is probably already tested by Devise team.

I think the only thing that needs to be tested is that an email is actually sent (that Devise is integrated properly). So I would create an integration test instead with one simple assertion `assert_emails 1`.

Reply:
Fair enough, good points provided. Creating the provided assertion in edit_user_test seemed logical because edit_user scope bundled with previous assertions seemed right. Recreated as integration test in last commit.